suite/tests/api/detach_state.c

```diff
@@ -462,22 +583,106 @@ DECL_EXTERN(safe_stack)
 #define FUNCNAME unique_values_to_registers
         DECLARE_FUNC(FUNCNAME)
 GLOBAL_LABEL(FUNCNAME:)
+#ifdef __AVX512F__
+#    define Z0 XMM0_LOW_BASE()
+#    define Z1 XMM0_HIGH_BASE()
+#    define Z2 YMMH0_LOW_BASE()
+#    define Z3 YMMH0_HIGH_BASE()
+#    define Z4 ZMMH0_0_BASE()
+#    define Z5 ZMMH0_1_BASE()
+#    define Z6 ZMMH0_2_BASE()
+#    define Z7 ZMMH0_3_BASE()
+        SET_ZMM_IMMED(0, Z0, Z1, Z2, Z3, Z4, Z5, Z6, Z7)
+        SET_ZMM_IMMED(1, Z0, Z1, Z2, Z3, Z4, Z5, Z6, Z7)
+        SET_ZMM_IMMED(2, Z0, Z1, Z2, Z3, Z4, Z5, Z6, Z7)
+        SET_ZMM_IMMED(3, Z0, Z1, Z2, Z3, Z4, Z5, Z6, Z7)
+        SET_ZMM_IMMED(4, Z0, Z1, Z2, Z3, Z4, Z5, Z6, Z7)
+        SET_ZMM_IMMED(5, Z0, Z1, Z2, Z3, Z4, Z5, Z6, Z7)
+        SET_ZMM_IMMED(6, Z0, Z1, Z2, Z3, Z4, Z5, Z6, Z7)
+        SET_ZMM_IMMED(7, Z0, Z1, Z2, Z3, Z4, Z5, Z6, Z7)
+        SET_ZMM_IMMED(8, Z0, Z1, Z2, Z3, Z4, Z5, Z6, Z7)
```

Comment:
Ditto. (offline: apparently the test is not enabled for 32-bit, maybe comment instead).

Reply:
This one is not inside ifdef X64.  Added.